DetectorDescription/DDCMS/plugins/DDHCalAngular.cc

```diff
@@ -0,0 +1,74 @@
+#include "DD4hep/DetFactoryHelper.h"
+#include "DetectorDescription/DDCMS/interface/DDPlugins.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+//#define EDM_ML_DEBUG
+
+static long  algorithm(dd4hep::Detector& /* description */,
+                       cms::DDParsingContext& ctxt,
+                       xml_h e,
+                       dd4hep::SensitiveDetector& /* sens */) {
+
+  cms::DDNamespace      ns(ctxt,e,true);
+  cms::DDAlgoArguments  args(ctxt, e);
+  // Header section of original DDHCalAngular.h
+  int            n           = args.value<int>("n");
+  int            startCopyNo = args.value<int>("startCopyNo");
+  int            incrCopyNo  = args.value<int>("incrCopyNo");
+  double         rangeAngle  = args.value<double>("rangeAngle"); //Angular range
+  double         startAngle  = args.value<double>("startAngle"); //Start anle
+  double         shiftX      = args.value<double>("shiftX");     //x Shift
+  double         shiftY      = args.value<double>("shiftY");     //y Shift
+  double         zoffset     = args.value<double>("zoffset");    //z offset
+  dd4hep::Volume mother      = ns.volume(args.parentName());
+  dd4hep::Volume child       = ns.volume(args.value<std::string>("ChildName"));
+  // Increment
+  double         dphi        = rangeAngle/n;
+#ifdef EDM_ML_DEBUG
+  edm::LogVerbatim("HCalGeom") << "DDHCalAngular: Parameters for positioning::"
+			       << " n " << n << " Start, Range, Delta " 
+			       << ConvertTo( startAngle, deg ) << " " 
+			       << ConvertTo( rangeAngle, deg ) << " " 
+			       << ConvertTo( dphi, deg )
+			       << " Shift " << shiftX << ":" << shiftY
+			       << "\n Parent " << mother.name() 
+			       << "\tChild " << child.name() 
+			       << " NameSpace " << ns.name();
+#endif
+  double theta  = 90._deg;
+  int    copy   = startCopyNo;
+  double phi    = startAngle;
+  for (int ii=0; ii<n; ++ii) {
+    double phix   = ConvertTo( phi, deg );
+    double phiy   = phix + 90._deg;
+    int    iphi   = (phix > 0) ? int(phix+0.1) : int(phix-0.1);
+    if (iphi >= 360) iphi -= 360;
+    phix = iphi;
+    dd4hep::Rotation3D rotation;
+    if (iphi != 0) {
```

Comment:
For lines 44-45 and 48, you can see in PR #24509 how I handled nearly identical code. My approach in this case would be:
 if (phix >= 2._pi) phix -= 2._pi; for lines 44-45
 if (std::abs(phix) >= 1.0_deg) { for line 48

Is it important that your version rounds a phix of 0.9 degree to 1.0 degree, while my version considers 0.9 degree as being 0?

Reply:
In the new version, the reduction of phix if it is greater than 2 pi has been omitted. It needs to be added, as shown in my previous comment. Also, iphi is unnecessary and potentially confusing.